src/backend/distributed/deparser/deparse_function_stmts.c

```diff
@@ -789,7 +789,14 @@ static void
 AppendGrantOnFunctionFunctions(StringInfo buf, GrantStmt *stmt)
 {
 	ListCell *cell = NULL;
-	appendStringInfo(buf, " ON %s ", ObjectTypeToKeyword(stmt->objtype));
+
+	/*
+	 * The FUNCTION syntax works for plain functions, aggregate functions, and window
+	 * functions, but not for procedures; use PROCEDURE for those. Alternatively, use
+	 * ROUTINE to refer to a function, aggregate function, window function, or procedure
+	 * regardless of its precise type.
+	 */
```

Comment:
definitely needs test. 

And, I'm not %100 sure if ROUTINE covers aggs for all cases. Where did you find that information? Would be good to link here as well

```SQL
alter routine taxi set owner to onderkalaci;
ERROR:  "public.taxi" is an aggregate function
```

Reply:
The ROUTINE is only intended to be used in GRANT/REVOKE statements. I add tests.